src/Test/Utilities/Shared/Syntax/SourceUtilities.cs

```diff
@@ -0,0 +1,50 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+using Microsoft.CodeAnalysis.Text;
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Roslyn.Test.Utilities.Syntax
+{
+    internal sealed class RandomizedSourceText: SourceText
+    {
+        private char hash(int position)
+        {
+            // good enough "random looking" hasher via the xorshift128plus algorithm.
+            UInt64 state0 = (UInt64) 1;
+            UInt64 state1 = (UInt64) position;
+            unchecked
+            {
+                var s1 = state0;
+                var s0 = state1;
+                state0 = s0;
+                s1 ^= s1 << 23;
+                s1 ^= s1 >> 17;
+                s1 ^= s0;
+                s1 ^= s0 >> 26;
+                state1 = s1;
```

Comment:
The "reason" is that I naively copied pseudocode from a wikipedia article.

I've cleaned up the locals and also changed state0 and state1 to be better chosen "constants".